assets/src/edit-story/components/movable-element/index.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * External dependencies
+ */
+import Moveable from 'react-moveable';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useState, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../app';
+import { getComponentForType } from '../../elements';
+
+const Element = styled.div`
+	cursor: pointer;
+	user-select: none;
+`;
+
+const MovableElement = ( props ) => {
+	const {
+		rotationAngle,
+		x,
+		y,
+		type,
+		selected,
+		id,
+		rest,
+	} = props;
+
+	const [ targetEl, setTargetEl ] = useState( false );
+
+	const moveable = useRef();
+
+	const {
+		actions: { setPropertiesOnSelectedElements, setPropertiesById, selectElementById, toggleElementIdInSelection },
+	} = useStory();
+
+	const frame = {
+		translate: [ 0, 0 ],
+		rotate: rotationAngle,
+	};
+
+	const setStyle = ( target ) => {
+		target.style.transform = `translate(${ frame.translate[ 0 ] }px, ${ frame.translate[ 1 ] }px) rotate(${ frame.rotate }deg)`;
+	};
+
+	const resetMoveable = ( target ) => {
+		frame.translate = [ 0, 0 ];
+		setStyle( target );
+		if ( moveable.current ) {
+			moveable.current.updateRect();
+		}
+	};
+
+	const handleSelectElement = useCallback( ( elId, evt ) => {
+		if ( evt.metaKey ) {
+			toggleElementIdInSelection( elId );
+		} else {
+			selectElementById( elId );
+		}
+		evt.stopPropagation();
+	}, [ toggleElementIdInSelection, selectElementById ] );
+
+	const comp = getComponentForType( type );
+	const Comp = comp; // why u do dis, eslint?
+
+	return (
+		<>
+			<Element
+				key={ id }
+				onClick={ ( evt ) => handleSelectElement( id, evt ) }
+			>
+				<Comp { ...rest } forwardedRef={ setTargetEl } />
+			</Element>
+			{ targetEl && <Moveable
```

Comment:
Are we doing a `Moveable` per element? How would we then do grouped transformations?

Reply:
Grouped transformation would have a separate `Moveable`, however, if we can use just `onPointerDown` and won't have to display multiple `Moveable`-s at the same time, then we can move back to using just one `Moveable` for now,